src/ripple/app/ledger/LedgerMaster.cpp

```diff
@@ -959,7 +947,7 @@ class LedgerMasterImp
                     WriteLog (lsTRACE, LedgerMaster) << "tryAdvance discovered missing " << missing;
                     if ((missing != RangeSet::absent) && (missing > 0) &&
                         shouldAcquire (mValidLedgerSeq, ledger_history_,
-                            ledger_history_index_, missing) &&
+                            getApp ().getSHAMapStore ().getCanDelete (), missing) &&
```

Comment:
Agreed. SHAMapStoreImp and LedgerMaster have several prior cyclic dependencies. I would recommend a new separate task to tackle the work.